parquet/tests/simple_variant_integration.rs

```diff
@@ -0,0 +1,1253 @@
+                if let Some(start_quote) = trimmed.rfind('"') {
+                    let before_quote = &trimmed[..start_quote];
+                    if let Some(second_quote) = before_quote.rfind('"') {
+                        case.variant_description =
+                            Some(before_quote[second_quote + 1..].to_string());
+                    }
+                }
+            }
+        } else if trimmed == "}, {" || trimmed == "}" {
+            // End of current case
+            if let Some(mut case) = current_case.take() {
+                if !case.parquet_file.is_empty()
+                    && (case.variant_file.is_some() || case.error_message.is_some())
+                {
+                    // Categorize the test based on its name if not already categorized
+                    if case.test_category == TestCategory::Primitives
+                        && case.error_message.is_none()
+                    {
+                        case.test_category = categorize_test(&case.test);
+                    }
+                    test_cases.push(case);
+                }
+            }
+        }
+    }
+
+    // Handle the last case if the JSON doesn't end with }, {
+    if let Some(mut case) = current_case {
+        if !case.parquet_file.is_empty()
+            && (case.variant_file.is_some() || case.error_message.is_some())
+        {
+            // Categorize the test based on its name if not already categorized
+            if case.test_category == TestCategory::Primitives && case.error_message.is_none() {
+                case.test_category = categorize_test(&case.test);
+            }
+            test_cases.push(case);
+        }
+    }
+
+    Ok(test_cases)
+}
+
+/// Categorize a test based on its test method name
+fn categorize_test(test_name: &Option<String>) -> TestCategory {
+    match test_name.as_ref().map(|s| s.as_str()) {
+        Some(name) if name.contains("Array") => TestCategory::Arrays,
+        Some(name) if name.contains("Object") => TestCategory::Objects,
+        Some(name) if name.contains("Unshredded") => TestCategory::SchemaValidation,
+        Some(name) if name.contains("Mixed") || name.contains("Nested") => TestCategory::Complex,
+        Some(name) if name.contains("Primitives") => TestCategory::Primitives,
+        _ => TestCategory::Primitives, // Default fallback
+    }
+}
+
+// Individual test functions with #[ignore] for progressive enablement
+// Following the exact pattern from the PR description
+
+#[test]
+#[ignore] // Enable once parquet-variant dependencies are added
+fn test_variant_integration_case_001() {
+    let harness = VariantIntegrationHarness::new().expect("Failed to create test harness");
+
+    let test_case = harness
+        .test_cases
+        .iter()
+        .find(|case| case.case_number == 1)
+        .expect("case-001 not found");
+
+    harness
+        .run_single_test(test_case)
+        .expect("case-001 should pass");
+}
```

Comment:
I think this pattern of creating one `#test` for each input case is very nice and results in easy to debug errors. However, since there will be so many cases (over 100 as I understand) I recommend we try and avoid as much repetition as possible

Could we write something like the following for each case? Then we could make a macro or something to generate all the functions

```suggestion
#[ignore] // Enable once parquet-variant dependencies are added
fn test_variant_integration_case_001() {
    test_harness().run_case(1)
}
```